### PR TITLE
Install LibreTime in vagrant env

### PR DESCRIPTION
**Pull request by hairmare**
Uses the old installer to get LibreTime installed quick and dirty on vagrant.

It uses the `install -fIap` command that does a local install and points the apache config
directly to the local working copy mounted in /vagrant.

While we don't have fancy autoloading for libretime like the docs do, this way
it's already easy to work against a local branch.

I'm not sure if the `-I` arg to the installer also covers the python parts of libretime.

* [x] update vagrant box to grab more memory (the default 512 MB results in oom kills) e394df83a5709459df61163f68b9982f0c2ee80d
* [x] solve port mapping dilemma bd2f96acf2cd76b26b8c47b6fa1408b354e2cf46
* [x] accessible icecast (it is still up to the user to enable it, an example is in #24)
* [x] virtual audio interface for testing line in/out features (works with pulseaudio out of the box)
* [x] more vagrant [docs](https://github.com/radiorabe/libretime/blob/feature/vagrant-install/docs/vagrant.md)

### CODE REVIEW

**Comment by hairmare** [comment]
This will really only be useable to hack on the php parts out of the box. analyzer needs some coaxing due to the config being based on 8080 from the webui and the actual apache running on 80.

I ended up hacking the apache config and letting the vhost listen on both ports to get the results up and running as a dev env. I'm not sure where analyzer is getting it's config from, it' doesn't seem to be `base_port` in `/etc/airtime/airtime.conf`.

**Comment by hairmare** [comment]
There external `8080` also somewhat conflicts with icecast running on `8080` internally. Does anyone have a preference when it comes to ports? Should we switch to a [private network ip](https://www.vagrantup.com/docs/networking/private_network.html)?

**Comment by Robbt** [comment]
I'm arbitrarily using 9080. A private network sounds fine as long as its documented, do we have docs for running vagrant to test anywhere ? 

Analyzer pulls the port from the conf after the web setup is ran so I think it is /etc/airtime/airtime.conf, at least based on my testing.

Does this write a default rabbitmq.ini file ?

**Comment by hairmare** [comment]
Some docs are on [getting started](https://github.com/LibreTime/libretime/blob/master/docs/index.md#getting-started) in the main docs page.

I think 9080 might be a sensible choice, at least locally I have nothing there most of the time. Analyzer does pull the port from the config, the problem is with this install the port gets NATed `8080 -> 80` which leads to the setup UI writing to `8080` to the file per default. I'll need to do some testing with private and/or public networks to figure out which one works for us.

All this does is call `install -fIap`, so no it doesn't fix the rabbitmq file issue.

**Comment by hairmare** [comment]
This should now install the whole shebang including virtual audio and expose all relevant ports in a straightforward way ready for devs.

The [docs](https://github.com/radiorabe/libretime/blob/2350650e2624546ea934727b354783d2a2ec86d7/docs/index.md) have been updated to reflect what ports to expect after install. You might have to `vagrant reload` to get audio working after the initial install (and LibreTime still needs #44 and #40 and similar fixes, I tested audio out with `mpg123` on the cli).

File: install [hunk 1]
```diff
@@ -414,12 +434,22 @@ if [ "$apache" = "t" ]; then
...
-            sed -e "s@WEB_ROOT@${web_root}@g" ${SCRIPT_DIR}/installer/apache/airtime-vhost-2.4 > /etc/apache2/sites-available/${airtimeconfigfile}
-        else
-            sed -e "s@WEB_ROOT@${web_root}@g" ${SCRIPT_DIR}/installer/apache/airtime-vhost > /etc/apache2/sites-available/${airtimeconfigfile}
+        apache_template_file=${SCRIPT_DIR}/installer/apache/airtime-vhost-2.4
+        if [ "$apacheversion" = "1" ]; then
+            # fall back to apache 1 config
+            apache_template_file=${SCRIPT_DIR}/installer/apache/airtime-vhost
         fi
+        sed \
+            -e "s@WEB_PORT_LISTEN@Listen ${listen_port}@g" \
```
Comment (hairmare) [inline install]: This will break installs on the regular port 😞

**Comment by hairmare** [comment]
I just changed the milestone on this after installing it one final time on a clean working copy. This has been my php dev install since I created it and it finally has working playout thanks to 0e8d0868e8eae8e7f245ee82fbae41652853e5bb.

The whole install process works through setup and the welcome guide through to having a red on-air button and listening to an "airtime pro" stream through the stream player.

**Comment by Robbt** [comment]
It works. Awesome.

**Pull request merged**